src/redux/sagas/tdexSaga.ts

```diff
@@ -48,46 +50,38 @@ function* fetchMarkets({ payload }: { payload: TDEXProvider }) {
   }
 }
 
-function* restoreProviders() {
+function* providersToRestore(): SagaGenerator<TDEXProvider[], TDEXProvider[]> {
   try {
-    // restore the providers from storage
-    const providers: TDEXProvider[] = yield call(getProvidersFromStorage);
-    for (const p of providers) {
-      yield put(addProvider(p));
-    }
-    // restore from registry in mainnet
-    try {
-      if (network.chain === 'liquid') {
-        const providersFromRegistry: TDEXProvider[] = yield call(getProvidersFromTDexRegistry);
-        const filteredProviders = providersFromRegistry.filter(
-          (prov: TDEXProvider) => providers.find((p) => p.endpoint === prov.endpoint) === undefined
-        );
-        for (const p of filteredProviders) {
-          yield put(addProvider(p));
-        }
-      } else {
-        if (providers.find((p) => p.endpoint === defaultProvider.endpoint) === undefined) {
-          yield put(
-            addProvider({
-              name: 'Default provider',
-              endpoint: defaultProvider.endpoint,
-            })
-          );
-        }
-      }
-    } catch (e) {
-      console.error(e);
-      yield put(addErrorToast(TDEXRegistryError));
+    // try to get providers from storage
+    const providersFromStorage = yield call(getProvidersFromStorage);
+    if (providersFromStorage.length > 0) {
+      return providersFromStorage;
     }
+  } catch {
+    yield put(addErrorToast(FailedToRestoreProvidersError));
+  }
+
+  // try to fetch providers from registry only on liquid
+  try {
+    const providersFromRegistry = yield call(getProvidersFromTDexRegistry);
```

Comment:
why did you remove the check for mainnet network? shouldn't this be done only in mainnet?

Reply:
U right, I've reverted this change